tests/static_map/static_map_test.cu

```diff
@@ -489,8 +488,43 @@ TEMPLATE_TEST_CASE_SIG("Shared memory static map",
                                d_keys_exist.data().get(),
                                d_keys_and_values_correct.data().get());
 
-    REQUIRE(none_of(d_keys_exist.begin(), d_keys_exist.end(), [] __device__(const bool key_found)
-{ return key_found;
+    REQUIRE(none_of(d_keys_exist.begin(), d_keys_exist.end(), [] __device__(const bool key_found) {
+      return key_found;
     }));
   }
 }
+
+template <typename K, typename V, std::size_t N>
+__global__ void shared_memory_hash_table_kernel(bool* key_found)
+{
+  namespace cg   = cooperative_groups;
+  using map_type = typename cuco::static_map<K, V, cuda::thread_scope_block>::device_mutable_view;
+  using find_map_type = typename cuco::static_map<K, V, cuda::thread_scope_block>::device_view;
+  __shared__ typename map_type::slot_type slots[N];
+  auto map = map_type::make_from_uninitialized_slots(cg::this_thread_block(), &slots[0], N, -1, -1);
+
+  auto g            = cg::this_thread_block();
+  std::size_t index = threadIdx.x + blockIdx.x * blockDim.x;
+  int rank          = g.thread_rank();
+
+  // insert {thread_rank, thread_rank} for each thread in thread-block
+  map.insert(cuco::pair<int, int>(rank, rank));
+  g.sync();
+
+  auto find_map       = find_map_type(slots, N, -1, -1);
```

Comment:
I've been meaning to add an explicit conversion operator from `device_mutable_view` to `device_view` to avoid having to do manual construction like this. Do you want to add that in this PR?

Reply:
Yep, I'll add it

Reply:
To understand clearly, in `device_mutable_view`, we would like to have an explicit conversion operator of the signature:
```    __host__ __device__ explicit operator device_view() noexcept;```. Is this possible since `device_view` hasn't been declared yet? Or should I overload a constructor and assignment operator for `device_view`?

Reply:
>we would like to have an explicit conversion operator of the signature:
>__host__ __device__ explicit operator device_view() noexcept;. Is this possible since device_view hasn't been declared yet?

Yes, you should be able to just forward declare either one.

Reply:
Another issue that I'm facing here is `get_slots()` is only defined in the device context for `device_view_base`. While I can modify that, is there a reason for it not having a host-side API? Would a host-side only API, if to be added, go to `static_map` instead?

Reply:
Keep it in `device_view_base` but make it host/device.